### DISCUSSION TITLE
Code: 'ERR_INVALID_URL' for every version from 1.116 up until latest 1.120.2

### DISCUSSION THREAD

**Discussion by DuckersMcQuack** [immich-app/immich#14182] [Q&A]
### The bug

Every version from 1.116 till 1.120.2 breaks this way, and all versions between 1.105.1 being the version i installed when i got the docker and the semi functional 1.110.0 can see thumbnails of images, but upon clicking, it can't read raw image data at all Just spitting errors.

Is it breaking because images and facial recognition was done with cuda that it fails the way it does? As i do not have that gpu installed anymore due to only one pcie x16 usable, which now is occupied by a sas card.
And is there a way to redo image scan without facial recognition while keeping image location and albums/album information?

```
[Nest] 708  - 11/17/2024, 2:49:09 AM     LOG [Api:EventRepository] Initialized websocket server
[Nest] 708  - 11/17/2024, 2:49:10 AM   ERROR [Api:An instance of SystemConfigDto has failed the validation:
 - property server.externalDomain has failed the following constraints: isUrl 
] Validation error
TypeError: Invalid URL
    at new URL (node:internal/url:806:29)
    at buildConfig (/app/immich/server/dist/utils/config.js:118:40)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /app/immich/server/dist/utils/config.js:53:26 {
  code: 'ERR_INVALID_URL',
  input: 'personaldomain.com'
}
api worker exited with code 1
Starting api worker
```
this error loops on startup on latest update.

### The OS that Immich Server is running on

Unraid 6.12.11

### Version of Immich Server

1.105.1 - 1.120.2

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Don't know. Installed via unraid apps.
```

### Your .env content

```Shell
ghcr.io/imagegenius/immich:1.105.1-1.110.0 and 1.110.0 to latest 1.120.2

Don't know what else to input here.
```

### Reproduction steps

1. install 1.105.1 with nvidia cuda for machine learning facial recognition, and start scanning images
2.Install latest version, see it bootlooping with 1.120.2, downgrade to 1.110.0, it can now see images in albums, but clicking on images will throw errors.

**Comment by alextran1502** [comment]
From the working version of 1.110 or 1.115, go to Administration > Settings > External Url and make sure the content has either http:// or https://

**Reply by DuckersMcQuack** [reply]
> From the working version of 1.110 or 1.115, go to Administration > Settings > External Url and make sure the content has either http:// or https://

What could be causing the issue on any versions from 1.116 and above? Or does this mean i'd have to start from scratch and re-import everything?

And what content are we talking about? 

Here's the error from the 1.115.0 version.

```
Starting microservices worker
[Nest] 669  - 11/17/2024, 3:33:45 PM     LOG [Microservices:EventRepository] Initialized websocket server
[Nest] 669  - 11/17/2024, 3:33:45 PM    WARN [Microservices:DuplicateService] Unknown keys found: {
  "image": {
    "preview": {
      "quality": 100
    },
    "thumbnail": {
      "size": 480,
      "quality": 100
    }
  }
}
[Nest] 669  - 11/17/2024, 3:33:45 PM   ERROR [Microservices:An instance of SystemConfigDto has failed the validation:
 - property server.externalDomain has failed the following constraints: isUrl 
] Validation error
[Nest] 669  - 11/17/2024, 3:33:45 PM     LOG [Microservices:SystemConfigService] LogLevel=log (set via system config)
[Nest] 669  - 11/17/2024, 3:33:45 PM    WARN [Microservices:DuplicateService] Unknown keys found: {
  "image": {
    "preview": {
      "quality": 100
    },
    "thumbnail": {
      "size": 480,
      "quality": 100
    }
  }
}
[Nest] 669  - 11/17/2024, 3:33:45 PM   ERROR [Microservices:An instance of SystemConfigDto has failed the validation:
 - property server.externalDomain has failed the following constraints: isUrl 
] Validation error
[Nest] 669  - 11/17/2024, 3:33:45 PM    WARN [Microservices:DuplicateService] Unknown keys found: {
  "image": {
    "preview": {
      "quality": 100
    },
    "thumbnail": {
      "size": 480,
      "quality": 100
    }
  }
}
[Nest] 669  - 11/17/2024, 3:33:45 PM   ERROR [Microservices:An instance of SystemConfigDto has failed the validation:
 - property server.externalDomain has failed the following constraints: isUrl 
] Validation error
[Nest] 669  - 11/17/2024, 3:33:45 PM     LOG [Microservices:MapRepository] Initializing metadata repository
[11/17/24 15:33:48] INFO     Started server process [706]                       
[11/17/24 15:33:48] INFO     Waiting for application startup.                   
[11/17/24 15:33:48] INFO     Created in-memory cache with unloading after 300s  
                             of inactivity.                                     
[11/17/24 15:33:48] INFO     Initialized request thread pool with 4 threads.    
[11/17/24 15:33:48] INFO     Application startup complete.                      
```
Or is this the http/https conflicts?


If this was the setting you meant, i can't turn it off. complains url is wrong.
![image](https://github.com/user-attachments/assets/ee5dce44-5966-4458-93d3-cb017e5809fd)
